### PR TITLE
[REFERENT] Add TerritorialCouncil members list

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: src/Repository/TerritorialCouncil/TerritorialCouncilMembershipRepository.php [hunk 1]
```diff
@@ -69,4 +74,82 @@ public function findAvailableMemberships(Candidacy $candidacy, SearchAvailableMe
...
+            ->select('COUNT(1)')
+            ->getQuery()
+            ->getSingleScalarResult()
+        ;
+    }
+
+    private function createQueryBuilderWithReferentTagsCondition(array $referentTags): QueryBuilder
+    {
+        return $this->createQueryBuilder('tcm')
+            ->leftJoin('tcm.territorialCouncil', 'territorialCouncil')
```
Comment (ottaviano) [inline src/Repository/TerritorialCouncil/TerritorialCouncilMembershipRepository.php:96]: `innerJoin` is better here

File: src/Repository/TerritorialCouncil/TerritorialCouncilMembershipRepository.php [hunk 2]
```diff
@@ -69,4 +74,82 @@ public function findAvailableMemberships(Candidacy $candidacy, SearchAvailableMe
...
+            ->leftJoin('territorialCouncil.referentTags', 'referentTag')
```
Comment (ottaviano) [inline src/Repository/TerritorialCouncil/TerritorialCouncilMembershipRepository.php:97]: same `innerJoin`

**Comment by illusionOfParadise** [comment]
[ch12163]

File: src/TerritorialCouncil/Filter/MembersListFilter.php [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    }
+
+    public function addReferentTag(ReferentTag $referentTag): void
+    {
+        $this->referentTags[] = $referentTag;
+    }
+
+    public function removeReferentTag(ReferentTag $referentTag): void
+    {
+        foreach ($this->referentTags as $key => $tag) {
```
Comment (ottaviano) [inline src/TerritorialCouncil/Filter/MembersListFilter.php:78]: why do not use `$this->referentTags->removeElement` method?

File: src/TerritorialCouncil/Filter/MembersListFilter.php [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+     *
+     * @Assert\NotBlank
+     * @Assert\Choice(choices={"a", "d"})
+     */
+    private $order = 'a';
+
+    public function __construct(array $referentTags, string $subscriptionType)
+    {
+        $this->referentTags = $referentTags;
+        $this->subscriptionType = $subscriptionType;
```
Comment (ottaviano) [inline src/TerritorialCouncil/Filter/MembersListFilter.php:60]: unused property?

File: templates/referent/_layout.html.twig [hunk 5]
```diff
@@ -111,8 +111,8 @@
                             {% endif %}
 
                             {% if isReferent %}
-                                <li class="{{ is_active_route(app.request, 'app_territorial_council_referent_*') ? 'active' }}">
-                                    <a href="{{ path('app_territorial_council_referent_designations_list') }}">Instances</a>
+                                <li class="{{ is_active_route(app.request, 'app_territorial_council_referent_*') or is_active_route(app.request, 'app_referent_territorial_council_*') ? 'active' }}">
```
Comment (ottaviano) [inline templates/referent/_layout.html.twig:114]: maybe we can rename your route prefix
to not have this 2 prefixes:
```
'app_territorial_council_referent_*'
'app_referent_territorial_council_*'
```

File: templates/referent/territorial_council/members.html.twig [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+                                    <td>
+                                        {% set is_email_subscribed = filter.subscriptionType and filter.subscriptionType in adherent.subscriptionTypes %}
+                                        <span class="status status__{{ is_email_subscribed ? '1' : '2' }}">
+                                            {{ (is_email_subscribed ? 'common.adherent.subscribed' : 'common.adherent.unsubscribed')|trans }}
+                                        </span>
+                                    </td>
+                                </tr>
+                            {% else %}
+                                <tr>
+                                    <td colspan="9" class="text--center">
```
Comment (ottaviano) [inline templates/referent/territorial_council/members.html.twig:80]: `colspan="6"`

File: src/TerritorialCouncil/Filter/MembersListFilter.php (hunk 4 above)
Comment (illusionOfParadise) [reply src/TerritorialCouncil/Filter/MembersListFilter.php:60]: It's used here https://github.com/EnMarche/en-marche.fr/pull/4749/files#diff-c20f802f969a22bcec26c4f2b7616d0eR72

File: templates/referent/_layout.html.twig (hunk 5 above)
Comment (illusionOfParadise) [reply templates/referent/_layout.html.twig:114]: I need distinguish 2 spaces here : https://github.com/EnMarche/en-marche.fr/pull/4749/files#diff-eeeb9ccf7e8b0f98d1b9a2f967c75e5bR3
It's another submenu.
Capture d’écran 2020-08-31 à 16 59 19

File: src/TerritorialCouncil/Filter/MembersListFilter.php (hunk 3 above)
Comment (illusionOfParadise) [reply src/TerritorialCouncil/Filter/MembersListFilter.php]: We don't need this method, I have been removed it.

**Review by ottaviano — APPROVED** [review]

**Pull request merged**